org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/wizard/template/NewProjectWizardTemplateParameterPage.java

```diff
@@ -0,0 +1,170 @@
+ * later page he can eventually configure the project template further.
+ * 
+ * @author Arne Deutsch - Initial contribution and API
+ */
+public class NewProjectWizardTemplateParameterPage extends WizardPage {
+
+	private final AbstractProjectTemplate projectTemplate;
+	private final List<Pair<ProjectVariable, Control>> variableWidgets = new ArrayList<>();
+
+	private boolean inUserAction = true;
+
+	public NewProjectWizardTemplateParameterPage(AbstractProjectTemplate projectTemplate) {
+		super("NewProjectWizardTemplateParameterPage");
+		this.projectTemplate = projectTemplate;
+	}
+
+	@Override
+	public void createControl(Composite parent) {
+		Composite main = new Composite(parent, SWT.NONE);
+		main.setLayoutData(new GridData(SWT.FILL, SWT.FILL, true, true));
+		main.setLayout(new GridLayout(2, false));
+		for (ProjectVariable variable : projectTemplate.getVariables()) {
+			Label label = new Label(main, SWT.NONE);
+			label.setText(variable.getLabel());
+			label.setToolTipText(variable.getDescription());
+			label.setLayoutData(new GridData(SWT.LEFT, SWT.CENTER, false, false));
+			if (variable instanceof StringProjectVariable) {
+				StringProjectVariable stringVariable = (StringProjectVariable) variable;
+				Text text = new Text(main, SWT.SINGLE | SWT.BORDER);
+				variableWidgets.add(Tuples.create(stringVariable, text));
+				text.setLayoutData(new GridData(SWT.FILL, SWT.CENTER, true, false));
+				text.setText(stringVariable.getValue());
+				text.setToolTipText(variable.getDescription());
+				text.addModifyListener(new ModifyListener() {
+					@Override
+					public void modifyText(ModifyEvent e) {
+						stringVariable.setValue(text.getText());
+						update();
+					}
+
+				});
+			} else if (variable instanceof StringSelectionProjectVariable) {
+				StringSelectionProjectVariable selectionVariable = (StringSelectionProjectVariable) variable;
+				Combo combo = new Combo(main, SWT.READ_ONLY);
+				variableWidgets.add(Tuples.create(selectionVariable, combo));
+				combo.setLayoutData(new GridData(SWT.FILL, SWT.CENTER, true, false));
+				combo.setItems(selectionVariable.getPossibleValues());
+				combo.setText(selectionVariable.getValue());
+				combo.setToolTipText(variable.getDescription());
+				combo.addSelectionListener(new SelectionAdapter() {
+					@Override
+					public void widgetSelected(SelectionEvent e) {
+						selectionVariable.setValue(combo.getText());
+						update();
+					}
+				});
+			} else if (variable instanceof BooleanProjectVariable) {
+				BooleanProjectVariable booleanVariable = (BooleanProjectVariable) variable;
+				Button checkbox = new Button(main, SWT.CHECK);
+				variableWidgets.add(Tuples.create(booleanVariable, checkbox));
+				checkbox.setLayoutData(new GridData(SWT.FILL, SWT.CENTER, true, false));
+				checkbox.setSelection(booleanVariable.getValue());
+				checkbox.setToolTipText(variable.getDescription());
+				checkbox.addSelectionListener(new SelectionAdapter() {
```

Comment:
use lambda

Reply:
Using a lambda here is not easaly possible because SelectionListener is not a functional interface. I will change all places where it is possible.